docs/criusupport.md

```diff
@@ -0,0 +1,99 @@
+* [2] https://openjdk.org/legal/assembly-exception.html
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH
+* Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# CRIU support
+
+**(Linux&reg; x86, Linux on POWER&reg; (Little Endian), and Linux on IBM Z&reg; only)**
+
+:fontawesome-solid-pencil:{: .note aria-hidden="true"} **Note:** Support for the Checkpoint/Restore In Userspace (CRIU) tool is currently provided as a technical preview in container environments. CRIU support is available for CRIU version packaged with Semeru package. This preview is supported for use in production environments, however, all APIs and command-line options are subject to change.
+
+In OpenJ9, the CRIU support includes an API that you can use to stop the VM at a checkpoint, save its state, and then run the VM from the point where it was stopped. The period of time between when the VM starts and when the application takes a checkpoint is referred to as the *checkpoint phase*. The application writes the VM state that was saved at the checkpoint to one or more image files. The saved state includes memory pages, methods, file systems, processes, and network connections. You can restore the VM from these files multiple times from the same point and in multiple environments.
+
+Because you can move checkpoint images between systems, they should not contain cryptographically-sensitive data. The CRIU technical preview introduces the `CRIUSECProvider` security provider, which provides the following limited set of security services: `MessageDigest` and `SecureRandom`. When you enable CRIU support, all existing security providers are removed from the security provider list and `CRIUSECProvider` is added. When you restore the VM, `CRIUSECProvider` is removed from the security provider list and the previous security providers are added back again.
+
+When the VM starts from the saved point instead of the beginning, the VM startup time improves and hence results in better VM performance.
+
+## Enabling CRIU support
+
+CRIU support is not enabled by default. You must enable it by specifying the [`-XX:+EnableCRIUSupport`](xxenablecriusupport.md) command-line option when you start your application.
+
+## Configuring CRIU support
+
+You can access the OpenJ9 CRIU support capabilities by specifying different options. The VM enables the addition of VM options on restore through the `CRIUSupport.registerRestoreOptionsFile` API and environment variables through the `CRIUSupport.registerRestoreEnvVariables` API. `OPENJ9_RESTORE_JAVA_OPTIONS` is a special environment variable for adding the VM options on restore.
+
+There are new options that work at checkpoint as well as at restore and some new options that work at restore only. There are also some existing options that work on restore but some of them behave differently.
+
+### New options
+
+You can use the following options during the checkpoint phase and when you restore the VM:
```

Comment:
`-XX:CheckpointGCThreads` has no use at restore time, its specifically for the checkpoint phase. Maybe this heading should be changed for clarity, but not sure what it means for the other options.

Reply:
@tajila - Please refer to @RSalman's comment. Would like to have your feedback for the same. Thanks!

Reply:
@tajila - How about the other 2 options mentioned `-XX:[+|-]CRIURestoreNonPortableMode` and `-Dorg.eclipse.openj9.criu.ImmutableEnvVars` Are these options used during the checkpoint phase and when you restore the VM? Thanks!

Reply:
those two are only used at JVM startup (checkpoint phase)

Reply:
Ok, thanks! So I will change the lead in line to "You can use the following options only during the checkpoint phase"